irohad/ametsuchi/impl/postgres_command_executor.cpp

```diff
@@ -1529,67 +1551,77 @@ namespace iroha {
-        const shared_model::interface::RemoveSignatory &command) {
+        const shared_model::interface::RemoveSignatory &command,
+        const shared_model::interface::types::AccountIdType &creator_account_id,
+        bool do_validation) {
       auto &account_id = command.accountId();
       auto &pubkey = command.pubkey().hex();
 
       StatementExecutor executor(remove_signatory_statements_,
-                                 do_validation_,
+                                 do_validation,
                                  "RemoveSignatory",
                                  perm_converter_);
-      executor.use("creator", creator_account_id_);
+      executor.use("creator", creator_account_id);
       executor.use("target", account_id);
       executor.use("pubkey", pubkey);
 
       return executor.execute();
     }
 
     CommandResult PostgresCommandExecutor::operator()(
-        const shared_model::interface::RevokePermission &command) {
+        const shared_model::interface::RevokePermission &command,
+        const shared_model::interface::types::AccountIdType &creator_account_id,
+        bool do_validation) {
       auto &permittee_account_id = command.accountId();
       auto revoked_perm = command.permissionName();
 
       StatementExecutor executor(revoke_permission_statements_,
-                                 do_validation_,
+                                 do_validation,
                                  "RevokePermission",
                                  perm_converter_);
-      executor.use("creator", creator_account_id_);
+      executor.use("creator", creator_account_id);
       executor.use("target", permittee_account_id);
       executor.use("revoked_perm", revoked_perm);
 
       return executor.execute();
     }
 
     CommandResult PostgresCommandExecutor::operator()(
-        const shared_model::interface::SetAccountDetail &command) {
+        const shared_model::interface::SetAccountDetail &command,
+        const shared_model::interface::types::AccountIdType &creator_account_id,
+        bool do_validation) {
       auto &account_id = command.accountId();
       auto &key = command.key();
       auto &value = command.value();
-      if (creator_account_id_.empty()) {
-        // When creator is not known, it is genesis block
-        creator_account_id_ = "genesis";
-      }
       std::string json_value = makeJsonString(value);
 
       StatementExecutor executor(set_account_detail_statements_,
-                                 do_validation_,
+                                 do_validation,
                                  "SetAccountDetail",
                                  perm_converter_);
-      executor.use("creator", creator_account_id_);
+      if (not creator_account_id.empty()) {
+        executor.use("creator", creator_account_id);
+      } else {
+        // When creator is not known, it is genesis block
+        const std::string genesis_creator_account_id = "genesis";
```

Comment:
please make this object outlive this call:
https://github.com/hyperledger/iroha/blob/a7914629851075cbd3de7ff8cfd5089f9a15e64d/irohad/ametsuchi/impl/postgres_command_executor.cpp#L1629